### PR TITLE
better GuardDisable

### PR DESCRIPTION
**Pull request by xiaoshihou514**

### CODE REVIEW

**Comment by glepnir** [comment]
maybe we don't need this `blacklist` table just use something like this

```lua
local ok, data = pcall(vim.api.nvim_get_autocmds,{group ='Guard',event = 'FileType', pattern = 'lua' })
if ok then
    vim.api.nvim_del_autocmd(data.id)
end
```

File: lua/guard/init.lua [hunk 1]
```diff
@@ -35,67 +31,74 @@ local function get_fts_keys()
...
     local arg = opts.args
     local _, bufnr = pcall(tonumber, arg)
     if bufnr then
-      if bufnr == 0 then
-        bufnr = api.nvim_get_current_buf()
-      end
-      if vim.tbl_contains(blacklist.buf, bufnr) then
-        blacklist.buf = vim.tbl_filter(function(v) return v ~= bufnr end, blacklist.buf)
+      local _, data = pcall(api.nvim_get_autocmds, { group = group, event = 'BufWritePre', buffer = bufnr })
+      if vim.tbl_isempty(data) then
```
Comment (glepnir) [inline lua/guard/init.lua:95]: check pcall status instead of use tbl_isempty.

**Comment by xiaoshihou514** [comment]
Conflicts getting messy, closing for now.

**Comment by glepnir** [comment]
use git rebase then resolve conflict 😄

**Comment by xiaoshihou514** [comment]
Yeah, I will also try to make the code structure cleaner, some files are a bit cluttered now.

**Comment by glepnir** [comment]
thanks :)

**Pull request closed without merging**